Extensions/render/mirror.lisp

```diff
@@ -52,21 +50,93 @@
 	    (%create-mirror-image mirror (1+ width) (1+ height))
 	    nil)))))
 
-;;;
-;;;
-;;;
-
-(defmethod notify-image-updated ((mirror image-mirror-mixin) region)
+(defmethod %notify-image-updated ((mirror image-mirror-mixin) region)
   (with-slots (dirty-region) mirror
     (if dirty-region
 	(setf dirty-region (region-union dirty-region region))
 	(setf dirty-region region))))
 
-;;; utility function
+;;;
+;;;
+;;;
+
+(defmethod %draw-image :around ((mirror image-mirror-mixin) 
+				image x y width height x-dest y-dest clip-region)
+  (when (image-mirror-image mirror)
+    (with-slots (image-lock) mirror
+      (climi::with-lock-held (image-lock)
+	(call-next-method)))))
+
+(defmethod %fill-image-mask :around ((mirror image-mirror-mixin) 
+				     image-mask x y width height x-dest y-dest clip-region ink background foreground)
+  (when (image-mirror-image mirror)
+    (with-slots (image-lock) mirror
+      (climi::with-lock-held (image-lock)
+	(call-next-method)))))
+
+(defmethod %fill-image :around ((mirror image-mirror-mixin) 
+				x y width height ink background foreground)
+  (when (image-mirror-image mirror)
+    (with-slots (image-lock) mirror
+      (climi::with-lock-held (image-lock)
+	(call-next-method)))))
+
+(defmethod %draw-paths :around ((mirror image-mirror-mixin) paths transformation region ink background foreground)
+  (when (image-mirror-image mirror)
+    (with-slots (image-lock) mirror
+      (climi::with-lock-held (image-lock)
+	(call-next-method)))))
+
+(defmethod %draw-image ((mirror image-mirror-mixin)
+			image x y width height
+			dst-dx dst-dy
+			clip-region)
+  (let ((region #+nil(rgb-image-copy
+		 image
+		 (image-mirror-image mirror)
+		 :x x :y y :width width :height height :x-dst x-dest :y-dst y-dest
+		 :clip-region clip-region)
+		(copy-image
+		 (image-mirror-image mirror)
+		 image
+		 :x x :y y :width width :height height
+		 ;;:x-dst x-dest :y-dst y-dest
+		 :src-dx dst-dx
+		 :src-dy dst-dy
+		 ;;:clip-region clip-region
+		 )
+		))
```

Comment:
are these commented arguments valid, or just leftovers? also please close blocks in the same line, like `)))`